opal/mca/threads/qthreads/threads_qthreads_module.c

```diff
@@ -34,40 +29,82 @@ struct opal_tsd_key_value {
     opal_tsd_destructor_t destructor;
 };
 
-static int opal_main_thread;
-struct opal_tsd_key_value *opal_tsd_key_values = NULL;
-static int opal_tsd_key_values_count = 0;
+/* 0: uninitialized, 1: initialized. */
+static opal_atomic_lock_t opal_thread_self_key_lock = OPAL_ATOMIC_LOCK_INIT;
+static int opal_thread_self_key_init = 0;
```

Comment:
Nit: is there a reason to use `int` here instead of `bool`?

Reply:
Thank you. `bool` is used here in the revised version.